components/dialog/test/dialog-fullscreen.vdiff.js

```diff
@@ -0,0 +1,79 @@
+import '../dialog-fullscreen.js';
+import { expect, fixture, html, nextFrame } from '@brightspace-ui/testing';
+import { footer, general, long, tabs } from './dialog-shared-contents.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+
+function createDialog(opts) {
+	const defaults = { content: html`${general}${footer}`, noPadding: false };
+	const { content, noPadding, width } =  { ...defaults, ...opts };
+	return html`
+		<d2l-dialog-fullscreen title-text="Dialog Title" ?no-padding="${noPadding}" width="${ifDefined(width)}" opened>
+			${content}
+		</d2l-dialog-fullscreen>
+	`;
+}
+
+function dispatchFullscreenWithinEvent(elem, state) {
+	elem.dispatchEvent(new CustomEvent(
+		'd2l-fullscreen-within', { bubbles: true, composed: true, detail: { state: state } }
+	));
+}
+
+describe('dialog-fullscreen', () => {
+
+	[/*'native',*/ 'custom'].forEach((type) => {
+
+		describe(type, () => {
+			before(async() => {
+				window.D2L.DialogMixin.preferNative = type === 'native';
+			});
+
+			[
+				{ screen: 'wider', viewport: { width: 1400, height: 700 } },
+				{ screen: 'wide', viewport: { width: 800, height: 500 } },
+				{ screen: 'narrow', viewport: { width: 600, height: 500 } },
+				{ screen: 'landscape', viewport: { width: 600, height: 320 } }
+			].forEach(({ screen, viewport }) => {
+				describe(screen, () => {
+					[
+						{ name: 'opened', template: createDialog() },
+						{ name: 'opened-set-width', template: createDialog({ content: html`${tabs}${general}`, width: 1200 }) },
+						{ name: 'opened-set-width-below-min', template: createDialog({ content: html`${tabs}${general}`, width: 200 }) },
+						{ name: 'opened-set-width-above-max', template: createDialog({ content: html`${tabs}${general}`, width: 4000 }) },
+						{ name: 'rtl', rtl: true, template: createDialog() },
+					].forEach(({ name, template, rtl }) => {
+						it(name, async() => {
+							await fixture(template, { viewport, rtl });
+							await expect(document).to.be.golden();
+						});
+					});
+				});
+			});
+
+			describe('internal', () => {
+
+				[
+					{ name: 'no-footer-content', template: createDialog({ content: long }) },
+					{ name: 'no-padding', template: createDialog({ content: html`<div style="background-color: var(--d2l-color-citrine); height: 100%; width: 100%;">No padding!</div>${footer}`, noPadding: true }) },
+					{ name: 'horizontal-overflow', template: createDialog({ content: html`${tabs}${general}` }) },
+					{ name: 'scroll-bottom-shadow', template: createDialog({ content: html`${long}${footer}` }) },
+					{ name: 'scroll-top-shadow', template: createDialog({ content: html`${long}${footer}` }), action: elem => elem.querySelector('#bottom').scrollIntoView() },
```

Comment:
Was wondering why this resulted in a diff... so previously `#dialogLong` went to "Line 12" but your `${long}` has 13 lines. Same thing happened in the `dialog.vdiff.js` tests.

Reply:
Yeah I should've called that out, I figured it wasn't worth adding another option in `dialog-shared-contents.js` to export or creating a one-off `template` just to have one less line

Reply:
Cool yeah, just wanted to make sure you did it consciously.